Added cdt package and CAM algorithm.

dmachlanski opened the pull request:
Adding the CAM algorithm from the CDT package. There are more algorithms available within the package but I've personally tested only CAM.

Please let me know if there is anything important that I forgot to include in the commit.

felixleopoldo commented on workflow/rules/structure_learning_algorithms/cdt_cam/rule.smk:
This should be like
```
rule:
    name: module_name

```
and `"cdt_cam"` should be substituted for `module_name`.

felixleopoldo commented on workflow/rules/structure_learning_algorithms/cdt_cam/info.json:
```diff
@@ -0,0 +1,22 @@
+        "url": "https://github.com/FenTechSolutions/CausalDiscoveryToolbox/tree/master"
+    },
+    "docs_url": "https://fentechsolutions.github.io/CausalDiscoveryToolbox/html/causality.html#cam",
+    "papers": [
```
papers is not used anymore, it can be removed

felixleopoldo commented on workflow/scripts/structure_learning_algorithms/cdt.py:
```diff
@@ -0,0 +1,82 @@
+import time
+import signal
+
+@contextmanager
```
It shouldn't be necessary to define the timeout functions here. Maybe have a look at [causallearn_graphs](https://github.com/felixleopoldo/benchpress/blob/master/workflow/rules/structure_learning_algorithms/causallearn_grasp/script.py#L6) to see how it is handled

dmachlanski commented:
Hi Felix, I've addressed your comments in the latest commit. I've also changed the branch to 'dev' instead of 'master' as I think I missed that previously as well.

dmachlanski commented:
Would you like me to generate the docs for this one as well?

felixleopoldo commented:
Hi Damian, 
Nice, yeah if you could generate docs for this one as well it would be great.

dmachlanski commented:
Docs now generated.

felixleopoldo commented:
Thanks, I'm trying to build a Docker image for ARM64, waiting for [response](https://github.com/FenTechSolutions/CausalDiscoveryToolbox/issues/166).

felixleopoldo commented:
Hi Damian

Did you try the cdt_cam modue using "score": "linear" ?
